alphagov/paas-cf
# [#112503399] Implement isolation between system components and runners - ACLs for components

# What

We need to ensure that Diego cells only have access to the minimum set of components that they need, to further secure the deployed environment and to ensure that if other security mechanisms fail we do not fail open. We also implemented ACLs from the Infrastructure network as part of this story.

Additionally we have renumbered CF networks to fit into 20 bit subnets to make ACLs more readable and easier to implement.   

# How

This PR adds AWS Network ACLs to the CF components subnets to restrict the ports allowed access from the cell and infrastructure networks to the CF network. This is implemented as a number of Terraform manifests to create ACLs and rules within ACLs.

### How to test

* Deploy an environment as normal by following the instructions in the README.md 
* Confirm that smoke tests complete successfully during deployment.
* Confirm that Network ACLs have been created by checking in the AWS gui (In VPC->Network ACLS)
* Try pushing an app to be extra-sure that things are OK.
* Run the Acceptance Tests for extra brownie points.

### Notes

You may notice that some of the ACL rules are super-sets of other rules. (i.e. we have a rule which allows a wide port range, but also rules which allow individual ports within that range) - This is because the gorouter component will issue requests from a random port between 1024-65535, and the responses from the diego cell back to these ports must be permitted). The intention is to later move the gorouter component to it's own dedicated subnet, at which point these rules can be modified more easily than having to capture all of the individual ports once again.

### Who can review

Anyone on the team except @combor or myself
